metricbeat/module/elasticsearch/index/data_xpack.go

```diff
@@ -0,0 +1,332 @@
+	xpackSchema = s.Schema{
+		"uuid":      c.Str("uuid"),
+		"primaries": c.Dict("primaries", indexStatsSchema),
+		"total":     c.Dict("total", indexStatsSchema),
+	}
+
+	indexStatsSchema = s.Schema{
+		"docs": c.Dict("docs", s.Schema{
+			"count": c.Int("count"),
+		}),
+		"fielddata": c.Dict("fielddata", s.Schema{
+			"memory_size_in_bytes": c.Int("memory_size_in_bytes"),
+			"evictions":            c.Int("evictions"),
+		}),
+		"indexing": c.Dict("indexing", s.Schema{
+			"index_total":             c.Int("index_total"),
+			"index_time_in_millis":    c.Int("index_time_in_millis"),
+			"throttle_time_in_millis": c.Int("throttle_time_in_millis"),
+		}),
+		"merges": c.Dict("merges", s.Schema{
+			"total_size_in_bytes": c.Int("total_size_in_bytes"),
+		}),
+		"query_cache":   c.Dict("query_cache", cacheStatsSchema),
+		"request_cache": c.Dict("request_cache", cacheStatsSchema),
+		"search": c.Dict("search", s.Schema{
+			"query_total":          c.Int("query_total"),
+			"query_time_in_millis": c.Int("query_time_in_millis"),
+		}),
+		"segments": c.Dict("segments", s.Schema{
+			"count":                         c.Int("count"),
+			"memory_in_bytes":               c.Int("memory_in_bytes"),
+			"terms_memory_in_bytes":         c.Int("terms_memory_in_bytes"),
+			"stored_fields_memory_in_bytes": c.Int("stored_fields_memory_in_bytes"),
+			"term_vectors_memory_in_bytes":  c.Int("term_vectors_memory_in_bytes"),
+			"norms_memory_in_bytes":         c.Int("norms_memory_in_bytes"),
+			"points_memory_in_bytes":        c.Int("points_memory_in_bytes"),
+			"doc_values_memory_in_bytes":    c.Int("doc_values_memory_in_bytes"),
+			"index_writer_memory_in_bytes":  c.Int("index_writer_memory_in_bytes"),
+			"version_map_memory_in_bytes":   c.Int("version_map_memory_in_bytes"),
+			"fixed_bit_set_memory_in_bytes": c.Int("fixed_bit_set_memory_in_bytes"),
+		}),
+		"store": c.Dict("store", s.Schema{
+			"size_in_bytes": c.Int("size_in_bytes"),
+		}),
+		"refresh": c.Dict("refresh", s.Schema{
+			"total_time_in_millis": c.Int("total_time_in_millis"),
+		}),
+	}
+
+	cacheStatsSchema = s.Schema{
+		"memory_size_in_bytes": c.Int("memory_size_in_bytes"),
+		"evictions":            c.Int("evictions"),
+		"hit_count":            c.Int("hit_count"),
+		"miss_count":           c.Int("miss_count"),
+	}
+)
+
+func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, info elasticsearch.Info, content []byte) error {
+	err := json.Unmarshal(content, &indicesStruct)
+	if err != nil {
+		return err
+	}
+
+	clusterStateMetrics := []string{"metadata", "routing_table"}
+	clusterState, err := elasticsearch.GetClusterState(m.HTTP, m.HTTP.GetURI(), clusterStateMetrics)
+	if err != nil {
+		return err
+	}
+
+	for name, index := range indicesStruct.Indices {
+		event := mb.Event{}
+		indexStats, err := xpackSchema.Apply(index)
+		if err != nil {
+			continue
```

Comment:
log the error here? even if it is only on debug?

Reply:
For the x-pack path, we decided not to *report* the error (via `r mb.ReporterV2`) as it would end up in the `metricbeat-*` index instead of the `.monitoring-es-6-mb-*` index, where all the other data ends up.

However, I think you are referring to *logging* the error in the metricbeat log, right?

Reply:
Yes, I refer to logging the error :slightly_smiling_face:

Reply:
👍 will do that here and other places in this PR where you have noted.

BTW, all the other metricsets (with x-pack monitoring code paths) will need this too so I'm going to make their logging part of a separate PR.

Reply:
Ok